examples/llm/src/models/mosaic_gpt.py

```diff
@@ -133,18 +133,47 @@ def _attn_bias(self, device, dtype):
 
     def forward(self,
                 input_ids: torch.LongTensor,
-                key_padding_mask: Optional[torch.ByteTensor] = None):
+                attention_mask: Optional[torch.ByteTensor] = None,
+                return_dict: Optional[bool] = None,
+                output_attentions: Optional[bool] = None,
+                output_hidden_states: Optional[bool] = None,
+                move_unmasked_logits_to_end: bool = False):
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        if not return_dict:
+            raise NotImplementedError(
+                'return_dict False is not implemented yet for MosaicGPT')
+        if output_attentions:
+            raise NotImplementedError(
+                'output_attentions is not implemented yet for MosaicGPT')
+        if output_hidden_states:
+            raise NotImplementedError(
+                'output_hidden_states is not implemented yet for MosaicGPT')
+
+        if attention_mask is not None and attention_mask[:, 0].sum(
+        ) != attention_mask.shape[0] and self.training:
+            raise NotImplementedError(
+                'MosaicGPT does not support training with left padding.')
+
         S = input_ids.size(1)
+
         assert (
-            S <= self.cfg.max_seq_len
-        ), f'Cannot forward input with seq_len={S}, this model only supports seq_len<={self.cfg.max_seq_len}'
+            S <= self.config.max_seq_len
+        ), f'Cannot forward input with seq_len={S}, this model only supports seq_len<={self.config.max_seq_len}'
 
         tok_emb = self.transformer.wte(input_ids)  # type: ignore
         if self.alibi:
+            # TODO: account for left padding with alibi
```

Comment:
I think once you merge my pr, this should be taken care of.

Reply:
I agree.